src/main/java/gregtech/api/capability/impl/MultiblockRecipeLogic.java

```diff
@@ -103,6 +104,40 @@ protected IMultipleTankHandler getOutputTank() {
         return controller.getOutputFluidInventory();
     }
 
+    @Override
+    protected boolean canWorkWithInputs() {
+        MultiblockWithDisplayBase controller = (MultiblockWithDisplayBase) metaTileEntity;
+        if (controller instanceof RecipeMapMultiblockController) {
+            RecipeMapMultiblockController distinctController = (RecipeMapMultiblockController) controller;
+
+            if (distinctController.canBeDistinct() && distinctController.isDistinct()) {
+                boolean canWork = false;
+                if (invalidatedInputList.isEmpty()) {
+                    return true;
+                }
+                if (!metaTileEntity.getNotifiedFluidInputList().isEmpty()) {
+                    canWork = true;
+                    invalidatedInputList.clear();
+                    metaTileEntity.getNotifiedFluidInputList().clear();
```

Comment:
There was a slight logic change that I missed when I asked to introduce the `else` part of this check. That is that the notified item inputs are no longer always cleared, but are instead only cleared if there are no notified fluid inputs.

This logic change can be compensated for by clearing the notified Item Inputs here, or just removing the `else` part of this statement that was introduced, and move back to always creating the iterator. I should not review when I am tired.